### ISSUE TITLE
Website accessibility issues

### ISSUE THREAD

**Issue by acrymble** [programminghistorian/jekyll#1315]
I've run a few of our pages through website accessibility tests. Here are some of the issues that we should want to address:

- All lesson icons on the /lessons page should have a machine-readable description that is semantically meaningful. At present these images are linked to the lesson, but a screen reader wouldn't likely know what they are. Related to this, all images should have alternative text.
- We should declare the natural language of all pages for screen readers.
- Links should not directly target images. When an image is embedded within an HTML document, content providers are able to provide an alternate text equivalent through the <code>alt</code> attribute.  When images are directly targeted, the image will load in the current browser window without any further information available to users of assistive technology.
- When ARIA attributes (state, roles, and properties) are not used correctly assistive technology may not correctly function as expected.   Valid markup should always be used.

---
I will suggest the technical team takes a look at this, as it seems like most fixes are quite easy.

**Comment by arojascastro** [comment]
Great job! I know @jenniferisasi is interested in this topic.

**Comment by jenniferisasi** [comment]
Oh, yes! This is something I've been meaning to talk about but I have so many things on my plate that I was waiting to cross a few first. I have been implementing the rules I know, following W3C standards, and so, for instance, in our Voyant-Tools lesson, I told the author to include [this step by step](https://programminghistorian.org/es/lecciones/analisis-voyant-tools#3-guardar-archivo) in written format as an alternative to visual-gif form.

**Comment by mdlincoln** [comment]
These are all great points. I'm going to split these into separate tickets. One question @acrymble - can you provide the site that you used to test this:

>We should declare the natural language of all pages for screen readers.

I had believed that all(?) of our pages already declared their proper natural language, but perhaps some are missing this, or we haven't formatted it correctly. I need more detail.

**Comment by mdlincoln** [comment]
also

>When ARIA attributes (state, roles, and properties) are not used correctly assistive technology may not correctly function as expected. Valid markup should always be used.

This may be from our dropdown menus... I need to see the full accessibility report to diagnose it however.

**Comment by acrymble** [comment]
the warnings came from testing a few pages with this tool: https://www.webaccessibility.com/

**Comment by mdlincoln** [comment]
The language warning seems to be triggering erroneously - it's pointing to a `<script>` tag? All of our pages start with `<html lang={{ page.lang }}>`, so I believe this is addressed.

**Comment by walshbr** [comment]
Since you've split this into several tickets, can we close this @mdlincoln?
